src/main/java/org/eclipse/uprotocol/uri/validator/UriValidator.java

```diff
@@ -90,8 +90,10 @@ static boolean isEmpty(UUri uri) {
      * @return Returns true if URI is of type RPC.
      */
     static boolean isRpcMethod(UUri uri) {
-        Objects.requireNonNull(uri, "Uri cannot be null.");
-        return !isEmpty(uri) && uri.getResource().getName().contains("rpc") && (uri.getResource().hasInstance() && !uri.getResource().getInstance().trim().isEmpty() || (uri.getResource().hasId() && uri.getResource().getId() != 0));
+        uri = Objects.requireNonNullElse(uri, UUri.getDefaultInstance());
```

Comment:
if uri is null return false
why are we doing this uri = Objects.requireNonNullElse(uri, UUri.getDefaultInstance()); and wasting computer
you can use a terninanry if statement if you want a return in one line

Reply:
Fixed these